test/Index/Store/unit-pcm-dependency.swift

```diff
@@ -74,6 +74,7 @@ func test() {
 // FILE2-NOT: Unit |{{.*}}ClangModuleB
 // FILE2-NOT: Record
 // FILE2: Unit | user | ClangModuleA | {{.*}}ClangModuleA-{{[A-Z0-9]*}}.pcm | ClangModuleA-{{[A-Z0-9]*}}.pcm-
+// FILE2: Record | user | {{.*}}unit-pcm-dependency.swift.tmp/s2.swift | s2.swift-
```

Comment:
Uncommenting `visitModuleReference` results in this line being added to the output, but I'm not sure if that was supposed to happen

Reply:
That's expected, there is this line in the test:
```
echo 'import ClangModuleA' > %t/s2.swift
```
So `s2.swift` only contains an import and nothing else. Before your change there was no symbol reference recorded for that file so no record was generated for that file. After your change a record needs to get generated for the module symbol reference.

I'd recommend to remove `unit-pcm-dependency.swift.tmp/` from your check line, this is test infrastructure specific, not test specific.